File: Students/John/python/lab05.py
"""

all codes are written and created by John Robson Wed Mar 10, 2021

"""

from random import choice

eyes = [":", ";", "8"]
noses = ["-", "", "^"]
mouths = [")", "(", "P"]

i = 0
while i < 5:
    buf = choice(eyes) + choice(noses) + choice(mouths)
    print(buf)
    i += 1

#Vertical

cheeks = ["()","[]", "  "]
eyes = ["^^", "--", "**"]
mouths = ["_", ".", "o"]

buf = choice(cheeks)
buf = buf[0] + choice(eyes) + buf[1]
buf = buf[0:2] + choice(mouths) + buf[2::]

print(buf)

File: Students/John/javascript/lab05.js
/*

all codes are written and created by John Robson Fri Mar 19, 2021

*/

const eyes = [":", ";", "8", ">:", ":'", "X"];
const noses = ["", "-", "^"];
const mouths = [")", "(", "*", "$", "O", "P", "D", "C", "3", "/"];

const armsV = ["", "<>", "\\/"];
const cheeksV = ["", "()", "[]", "{}"];
const eyesV = ["-", "^", "o", "@", "v", "O", "u"];
const mouthsV = ["_", ".", "o"];

export function runLab05() {
  renApp(
    [
      { el: "h1", text: "Lab 05 - Emoticon Generator" },
      { el: "button", text: "Generate Emoticon", cl: "emoticon-btn" },
    ],
    $("main")
  );

  $(".emoticon-btn").onclick = generateEmoticon;
}

const generateEmoticon = () => {
  limit($$(".emo"), 5)
  const emoteH = rand(eyes) + rand(noses) + rand(mouths),
    aV = rand(armsV),
    cV = rand(cheeksV),
    eV = rand(eyesV),
    mV = rand(mouthsV);
  let emoteV = aV === "" ? "" : aV[0];
  emoteV += cV === "" ? "" : cV[0];
  emoteV += eV + mV + eV;
  emoteV += cV === "" ? "" : cV[1];
  emoteV += aV === "" ? "" : aV[1];
  renApp({ el: "p", cl: "emo", text: rand([emoteH, emoteV]) }, $("main"));
};
